Salesforce version upgrade [DOC-835] (#6149)
segmentio/segment-docs: src/connections/sources/catalog/cloud-apps/salesforce/index.md

If you've added and selected custom fields to an existing collection and would not like to use `SystemModstamp` to bring in historical data, [contact Segment Support](https://segment.com/help/contact/){:target="_blank"} to configure them to sync. You do not need to include the field names.

## Version upgrade changes

> On March 25, 2024, Segment will upgrade the Salesforce source integration from version 39.0 to Salesforce's latest API version 58.0. With the upgrade, there are [deleted fields](#deleted-fields) as well as [added fields](#added-fields). 

Reviewer: ```suggestion
On March 25, 2024, Segment will upgrade the Salesforce source integration from version 39.0 to Salesforce's latest API version 58.0. With the upgrade, there are [deleted fields](#deleted-fields) and [added fields](#added-fields). 
```
